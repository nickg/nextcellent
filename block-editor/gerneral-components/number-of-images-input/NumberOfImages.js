import { __ } from "@wordpress/i18n";

// Load external dependency.
import "./numberOfImages.scss";

/**
 *
 * @param value
 * @param onChange
 * @param props
 * @returns {JSX.Element}
 * @constructor
 */
function NumberOfImages({ type = "gallery", value, onChange, ...props }) {
	// Unique ID for the id.
	const inputId = `nextcellent-block-number-of-images`;

	// Function to handle the onChange event.
	const onChangeValue = (event) => {
		onChange(event.target.value);
	};

	// Return the fieldset.
	return (
		<div className="nextcellent-number-of-images">
			{/* Label for the input. */}
			<label htmlFor={inputId}>{__("Number of images", "nggallery")}</label>

			{/* Input field. */}
			<input
				id={inputId}
				type="number"
				min="0"
				step="1"
				value={value}
				onChange={onChangeValue}
			/>
			{type == "gallery" && (
				<p>
					{__(
						"The number of images before pagination is applied. Leave empty or 0 for the default from the settings.",
						"nggallery"
					)}
				</p>
			)}
			{type == "recent" && (
				<p>
					{__("The number of images that should be displayed.", "nggallery")}
				</p>
			)}
		</div>
	);
}

export default NumberOfImages;
