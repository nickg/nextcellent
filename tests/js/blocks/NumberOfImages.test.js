import {
	render, //test renderer
	cleanup, //resets the JSDOM
	fireEvent //fires events on nodes
 } from "@testing-library/react";
import NumberOfImages from '../../../block-editor/gerneral-components/numberOfImages/NumberOfImages'

describe("Number of images component", () => {
	afterEach(cleanup); //reset JSDOM after each test

	//It handles having no saved attribute
	it("matches snapshot", () => {
		expect(
		  render(
			<NumberOfImages
				value={""}
				onChange ={jest.fn()}
			/>
		  )
		).toMatchSnapshot();
	});

	it("calls the onChange function", () => {
		//mock function to test with
		const onChange = jest.fn();
		//Render component and get back getByLabelText()
		const {getByLabelText} = render(
		  <NumberOfImages
			onChange={onChange}
			value={""}
		  />
		);
		//Get the input by label text
		const input = getByLabelText('Number of images');

		fireEvent.input(input, {
		  target: { value: '0' }
		});

		expect(onChange).toHaveBeenCalledTimes(1);
	})

	it("passes the right value to onChange", () => {
		const onChange = jest.fn();
		const {getByLabelText} = render(
		  <NumberOfImages
			onChange={onChange}
			value={""}
		  />
		);
		const input = getByLabelText('Number of images');
		//Fire a change event on the input
		fireEvent.change(input, {
		  target: { value: 0 }
		});
		//Was the new value -- not event object -- sent?
		expect(onChange).toHaveBeenCalledWith("0");
	  });

	  it("does not accept text inputs", () => {
		const onChange = jest.fn();
		const {getByLabelText} = render(
		  <NumberOfImages
			onChange={onChange}
			value={""}
		  />
		);
		const input = getByLabelText('Number of images');

		fireEvent.change(input, {
		  target: { value: "Test" }
		});

		expect(onChange).not.toHaveBeenCalled();
	  })

	  it("accepts numbers as inputs", () => {
		const onChange = jest.fn();
		const {getByLabelText} = render(
		  <NumberOfImages
			onChange={onChange}
			value={""}
		  />
		);
		const input = getByLabelText('Number of images');

		fireEvent.change(input, {
		  target: { value: "0" }
		});

		expect(onChange).toHaveBeenCalledWith('0');

		fireEvent.change(input, {
		  target: { value: 0 }
		});

		expect(onChange).toHaveBeenCalledWith('0');
	  })
 })
