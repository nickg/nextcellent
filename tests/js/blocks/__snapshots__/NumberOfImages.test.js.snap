// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Number of images component matches snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="nextcellent-number-of-images"
      >
        <label
          for="nextcellent-block-number-of-images"
        >
          Number of images
        </label>
        <input
          id="nextcellent-block-number-of-images"
          min="0"
          step="1"
          type="number"
          value=""
        />
        <p>
          The number of images before pagination is applied. Leave empty or 0 for the default from the settings.
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="nextcellent-number-of-images"
    >
      <label
        for="nextcellent-block-number-of-images"
      >
        Number of images
      </label>
      <input
        id="nextcellent-block-number-of-images"
        min="0"
        step="1"
        type="number"
        value=""
      />
      <p>
        The number of images before pagination is applied. Leave empty or 0 for the default from the settings.
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
